homeassistant/components/onvif/camera.py

```diff
@@ -204,8 +204,8 @@ async def _async_get_stream_uri(self) -> str:
                 self._stream_uri_future.set_exception(err)
             raise
         url = URL(uri_no_auth)
-        url = url.with_user(self.device.username)
-        url = url.with_password(self.device.password)
+        url = url.with_user(self.device.username or None)
+        url = url.with_password(self.device.password or None)
```

Comment:
And if the password would be an empty string?

Reply:
Are `http://foo@example.com` and `http://foo:@example.com` not technically the same thing? I could potentially see some servers interpreting those differently, but seems unlikely.

I guess we could also have different behavior for empty string password vs. username. The current URL being:

```
>>> yarl.URL('http://example.com').with_user('').with_password('')
URL('http://:@example.com')
```

seems pretty broken to me. Another approach is we could *not* support empty string usernames but allow empty string passwords if username is non-empty. WDYT?

Reply:
> WDYT?

I think there is no good call in the choice you put out there. It mostly depends on the "great" implementations in the many cameras out there.

IMHO, it is better to actually properly handle None vs an empty string; and not just add a falsy "or None" in here and thus removing any of the controls.

Reply:
> Or maybe even a tick box user has to tick like:
> 
> * [ ]  _No authentication_

This is the only clean implementation I can see.

Reply:
Empty password might be a thing, but empty password + empty username makes no sense.

I really think we're overthinking this here. If we want the dead simple approach that fixes the problem, why don't we do something like:

```python
if self.device.username or self.device.password:
  url = url.with_user(self.device.username)
  url = url.with_password(self.device.password)
```

That solves the problem of sending `http://:@example.com` which is nonsense but leaves open for either to be empty. We'll send both if either are non-empty.

Reply:
I completely agree. @frenck, what do you think?